R/plot_carp.R

```diff
@@ -354,15 +440,90 @@ carp_dendro_plot <- function(x,
       if ( k > NROW(x$X) ) {
         crv_error(sQuote("k"), " cannot be more than the observations in the original data set (", NROW(x$X), ").")
       }
+
+      percent <- get_feature_paths(x, features = character()) %>% filter(.data$NCluster == k) %>%
+        select(.data$GammaPercent) %>%
+        summarize(NCluster = mean(.data$GammaPercent)) %>%
+        pull
     }
+    # heatmap
+    U <- get_clustered_data(x, percent = percent, refit = refit)
 
-    my.cols <- adjustcolor(base_colors, alpha.f = .2)
-    my.rect.hclust(as.hclust(x), k = k, border = 2, my.col.vec = my.cols, lwd = 3)
-  }
+    # dendrogram
+    c <- color_branches(d, k = k)
+    dend <- as.ggdend(rev(c))
+    segs <- dend$segments
 
-  invisible(x)
-}
+    adjust <- get_feature_paths(x, features = character()) %>% filter(.data$NCluster == 1) %>%
+      select(.data$GammaPercent) %>%
+      summarize(GammaPercent = min(.data$GammaPercent)) %>%
+      pull
+    segs$y <- segs$y*adjust
+    segs$yend <- segs$yend*adjust
+
+    tree <- rev(as.hclust(x))
```

Comment:
This block of code comes up a few times. Would it be better to put it in a helper function?